### PR TITLE
Create log files and redirect output to them

### PR DESCRIPTION
**Pull request by jsuhaas22**
The 3 commits in this PR do the following:

1. add or move around some helpful echo statements
2. create log files by timestamp of when the scripts were run
3. redirect all output to log files and show only echo statements

### CODE REVIEW

File: scripts/build_distro.sh [hunk 1]
```diff
@@ -23,17 +25,19 @@ build=$1
...
 
     echo "> Cleaning up ${build}"
-    tar --use-compress-program="pigz --best --recursive | pv" -cf tisdk-${distro}-${machine}-rootfs.tar.xz tisdk-${distro}-${machine}-rootfs
+    (tar --use-compress-program="pigz --best --recursive | pv" -cf tisdk-${distro}-${machine}-rootfs.tar.xz tisdk-${distro}-${machine}-rootfs) &>>"${LOG_FILE}"
+
     rm -rf tisdk-${distro}-${machine}-rootfs
 
-    tar --use-compress-program="pigz --best --recursive | pv" -cf tisdk-${distro}-${machine}-boot.tar.xz tisdk-${distro}-${machine}-boot
+    tar --use-compress-program="pigz --best --recursive | pv" -cf &>>"${LOG_FILE}"
+	tisdk-${distro}-${machine}-boot.tar.xz tisdk-${distro}-${machine}-boot
```
Comment (sskartheekadivi) [inline scripts/build_distro.sh:33]: why did we split the command? This won't work.

File: scripts/setup.sh [hunk 2]
```diff
@@ -18,10 +20,14 @@ function setup_build_tools() {
...
-        tar -Jxf gcc-arm-9.2-2019.12-x86_64-arm-none-linux-gnueabihf.tar.xz
-        rm gcc-arm-9.2-2019.12-x86_64-arm-none-linux-gnueabihf.tar.xz
-        echo "> Arm Toolchain: downloaded .." 
+        wget https://developer.arm.com/-/media/Files/downloads/gnu-a/9.2-2019.12/binrel/gcc-arm-9.2-2019.12-x86_64-arm-none-linux-gnueabihf.tar.xz &>>/dev/null
+		if [ $? -eq 0 ]; then
+			echo "> Arm Toolchain: downloaded .."
+			tar -Jxf gcc-arm-9.2-2019.12-x86_64-arm-none-linux-gnueabihf.tar.xz &>>"${LOG_FILE}"
+			rm gcc-arm-9.2-2019.12-x86_64-arm-none-linux-gnueabihf.tar.xz
+		else
+			echo "> Arm Toolchain: could not download"
```
Comment (sskartheekadivi) [inline scripts/setup.sh:29]: Better say Failed to download along with the return code to give a more clear perspective of what happened.

File: scripts/setup.sh [hunk 3]
```diff
@@ -33,10 +39,14 @@ function setup_build_tools() {
...
-        tar -Jxf gcc-arm-9.2-2019.12-x86_64-aarch64-none-linux-gnu.tar.xz
-        rm gcc-arm-9.2-2019.12-x86_64-aarch64-none-linux-gnu.tar.xz
-        echo "> Aarch64 Toolchain: downloaded .." 
+        wget https://developer.arm.com/-/media/Files/downloads/gnu-a/9.2-2019.12/binrel/gcc-arm-9.2-2019.12-x86_64-aarch64-none-linux-gnu.tar.xz &>>/dev/null
+		if [ $? -eq 0 ]; then
+			echo "> Aarch64 Toolchain: downloaded .." 
+			tar -Jxf gcc-arm-9.2-2019.12-x86_64-aarch64-none-linux-gnu.tar.xz  &>>"${LOG_FILE}"
+			rm gcc-arm-9.2-2019.12-x86_64-aarch64-none-linux-gnu.tar.xz
+		else
+			echo "> Aarch Toolchain: Could not download"
```
Comment (sskartheekadivi) [inline scripts/setup.sh:48]: same as above.

File: scripts/setup.sh [hunk 4]
```diff
@@ -1,30 +1,52 @@
...
+source ${topdir}/scripts/common.sh
+
+function setup_log_file() {
+	# create the log directory if it doesn't already exist
+	LOG_DIR="${topdir}/logs"
+	mkdir -p "${LOG_DIR}"
+
+	# we want to create a different log file for each run. So name these files
+	# with timestamps.
+	export LOG_FILE="$LOG_DIR/run_$(date +"%Y-%m-%d_%H:%M:%S").log"
```
Comment (sskartheekadivi) [inline scripts/setup.sh:12]: better to have build name with the log file instead or along with the timestamps.

File: scripts/build_bsp.sh [hunk 5]
```diff
@@ -175,10 +175,10 @@ machine=$1
     target_board=($(read_machine_config ${machine} atf_target_board))
 
     echo "> ATF: building .."
-    make -j`nproc` ARCH=aarch64 CROSS_COMPILE=aarch64-none-linux-gnu- PLAT=k3 TARGET_BOARD=${target_board} SPD=opteed
+	make -j`nproc` ARCH=aarch64 CROSS_COMPILE=aarch64-none-linux-gnu- PLAT=k3 TARGET_BOARD=${target_board} SPD=opteed &>>"${LOG_FILE}"
```
Comment (sskartheekadivi) [inline scripts/build_bsp.sh:178]: please do not use tabs. Use 4 spaces instead. :)

File: scripts/build_distro.sh [hunk 6]
```diff
@@ -1,16 +1,18 @@
...
 distro=$3
 
     hostname=($(read_machine_config ${machine} hostname))
-    bdebstrap \
+    (bdebstrap \
         -c ${topdir}/configs/bdebstrap_configs/${distro}.yaml \
         --name ${topdir}/build/${build} \
         --target tisdk-${distro}-${machine}-rootfs \
-        --hostname "${hostname}" -f
+		--hostname "${hostname}" -f) &>>"${LOG_FILE}"
```
Comment (sskartheekadivi) [inline scripts/build_distro.sh:15]: why does the whole command has to be enclosed inside parentheses?

**Review by sskartheekadivi — CHANGES_REQUESTED** [review]
Thanks for the PR. Please address the comments.

File: scripts/setup.sh (hunk 4 above)
Comment (jsuhaas22) [reply scripts/setup.sh:12]: In "sudo ./build.sh am62xx-evm", is the build name "am62xx-evm"? If yes, what should the name be if the user doesn't specify any arguments ("all"?)?

Comment (sskartheekadivi) [reply scripts/setup.sh:12]: the readme is outdated. My bad. 
It's not `sudo ./build.sh <machine>` anymore. It is `sudo ./build.sh <build>`.
The <build> is what you have in "builds.toml" file.
So I suggest log file to be `logs/${build}.log` or `logs/${build}_<timestamp>.log`

FYI, we'll not build all if no arg is specified. That will change in the README and build.sh update that I'm gonna do today. Coz no one will want to build all supported build configs.

The Github Workflow follows the same approach.

**Review by sskartheekadivi — APPROVED** [review]

**Pull request merged**